File: DataStructuresAndLibraries/pagina07/3171/3171.cpp
#include <iostream>
#include <map>
#include <vector>
#include <set>

using namespace std;

void DFS(set<int> visitados, map<int, vector<int>> grafico, int node){
    visitados.insert(node);
    for(const int& visinho: grafico[node]){
        DFS(visitados, grafico, visinho);
    }
}

int main(){

    int n_leds, n_segmentos;
    while(cin >> n_leds >> n_segmentos){

        map<int, vector<int>> grafico;
        set<int> visitados;

        for(int i = 0; i < n_segmentos; i++){
            int N, M;
            cin >> N >> M;

            if(grafico.find(N) == grafico.end()) grafico[N] = vector<int>();
            if(grafico.find(M) == grafico.end()) grafico[M] = vector<int>();

            grafico[N].push_back(M);
            grafico[M].push_back(N);
        }

        DFS(visitados, grafico, 1);

        cout << (n_leds == visitados.size() ? "COMPLETO" : "INCOMPLETO") << endl;
    }

    return 0;
}

File: DataStructuresAndLibraries/pagina07/3171/3171.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").split("\n").pop()

const DFS = (visitados, grafico, node) => {
    visitados.add(node)
    for(const vizinho of grafico[node]){
        DFS(visitados, grafico, vizinho)
    }
}

while (input.length > 0){

    const [n_leds, n_segmentos] = input.shift().split(" ").map(Number)
    
    let grafico = {}
    let visitados = new Set()

    for(let i = 0; i < n_segmentos; i++){
        const [N, M] = input.shift().split(" ").map(Number)

        if(!Object.keys(grafico).includes(N)) grafico[n] = []
        if(!Object.keys(grafico).includes(M)) grafico[n] = []

        grafico[N].push(M)
        grafico[M].push(N)
    }

    DFS(visitados, grafico, 1)

    console.log(visitados.size() == n_leds ? "COMPLETO" : "INCOMPLETO")

}